### PR TITLE
Docker Image build feature

### PR DESCRIPTION
**Pull request by akomic**
Implementation of build block for docker_image resource

### CODE REVIEW

File: scripts/compile.sh [hunk 1]
```diff
@@ -10,7 +10,7 @@ fi
 PROVIDER_NAME="docker"
 TARGET_DIR="$(pwd)/results"
 XC_ARCH=${XC_ARCH:-"386 amd64 arm"}
-XC_OS=${XC_OS:=linux darwin windows freebsd openbsd solaris}
+XC_OS=${XC_OS:=linux darwin windows openbsd solaris}
```
Comment (mavogel) [inline scripts/compile.sh:13]: What was the reason the remove `freebsd` here?

File: scripts/testacc_setup.sh [hunk 2]
```diff
@@ -17,7 +17,7 @@ openssl req \
 # Create auth
 mkdir -p "$(pwd)"/scripts/testing/auth
 # Start registry
-docker run --rm --entrypoint htpasswd registry:2 -Bbn testuser testpwd > "$(pwd)"/scripts/testing/auth/htpasswd
+docker run --rm --entrypoint htpasswd registry:2.6.2 -Bbn testuser testpwd > "$(pwd)"/scripts/testing/auth/htpasswd
```
Comment (mavogel) [inline scripts/testacc_setup.sh:20]: I assume the removal of the `htpasswd` binary made you pin the version? (https://github.com/docker/docker.github.io/issues/11060)

File: scripts/compile.sh (hunk 1 above)
Comment (protomonk) [reply scripts/compile.sh:13]: One of the "archive" depending packages is not compiling on FreeBSD with go >= 1.12. 

https://github.com/moby/moby/pull/38818

Wanted to verify before proceeding, is it acceptable to remove freebsd or should I try to find alternative.
There is a benefit in using docker/cli packages in having exactly the same way of behaviour as docker-cli ensuring compatibility (e.g. support .dockerignore).

Comment (akomic) [reply scripts/compile.sh:13]: One of the "archive" depending packages is not compiling on FreeBSD with go >= 1.12. 

https://github.com/moby/moby/pull/38818

Wanted to verify before proceeding, is it acceptable to remove freebsd or should I try to find alternative.
There is a benefit in using docker/cli packages in having exactly the same way of behaviour as docker-cli ensuring compatibility (e.g. support .dockerignore).

File: scripts/testacc_setup.sh (hunk 2 above)
Comment (akomic) [reply scripts/testacc_setup.sh:20]: Yes, looked like the least intrusive way to fix the testing.
Another alternative would be to alter test bash scripts to add apache2-utils before generating credentials.

File: scripts/compile.sh (hunk 1 above)
Comment (akomic) [reply scripts/compile.sh:13]: @mavogel what would you like me to do with this one?

Comment (mavogel) [reply scripts/compile.sh:13]: I'll reach out to the hashicorp folks and ask them. I'll come to to you here. 
IMHO it makes total sense to use `docker/cli` packages for reusing functionality

Comment (mavogel) [reply scripts/compile.sh:13]: @akomic sorry for the delay. As part of the terraform `0.13` migration this provider will be forked off to a _community_ maintained one, meaning we will have full control over how to build and release it to the terraform registry via GH actions. 
TL;DR: I'll merge all remaining PRs, including this one, and then fork off. IMHO we'd go with `docker/cli`

Comment (protomonk) [reply scripts/compile.sh:13]: @mavogel Great, there are some things I'd like to add/improve regarding this functionality but wanted to see if it is not futile. I'll continue with contribution once it's forked.

Thanks

Comment (akomic) [reply scripts/compile.sh:13]: @mavogel Great, there are some things I'd like to add/improve regarding this functionality but wanted to see if it is not futile. I'll continue with contribution once it's forked.

Thanks

**Comment by mavogel** [comment]
merged and fixed with https://github.com/terraform-providers/terraform-provider-docker/commit/250b58a9f87af56e2b5a7655355fc2f850b9ac52

Github does not get it 🤔
